### PR TITLE
Basic UI added, rest-part fixed.

### PR DESCRIPTION
**Pull request by svelaf71**

### CODE REVIEW

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * Method for mapping on default categories url
+     * shows the list of available categories and allows to add a new one
+     *
+     * @param userID
+     * @return
+     */
+    @GetMapping
+    public ModelAndView getPage(@RequestParam(value = "user_id", required = false) Integer userID) {
+        if (userID == null)
+            userID = 2;
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java]: what?

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        ModelAndView model = new ModelAndView("categories");
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java]: I see that someone uses ModelAndView, someone uses Model.

Please discuss this and use only one approach. I would like to use Model, as it is newer

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        final String uri = new String(restCategoryURL + userID);
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java]: Why need to create new String object?

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java [hunk 4]
```diff
@@ -47,16 +43,24 @@ public Category getById(int id) throws DataNotFoundException {
...
+        }
     }
 
-    public void update(Category category) {
-        categoryDAO.update(category);
+    public void update(Category category) throws DataNotFoundException {
+        if (category.getName() != null && category.getUser() != null) {
+            categoryDAO.update(category);
+        } else {
+            throw new DataNotFoundException("Incorrect fields for category");
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java]: IllegalArgumentException is better here

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java [hunk 5]
```diff
@@ -47,16 +43,24 @@ public Category getById(int id) throws DataNotFoundException {
         if (category != null) {
             return category;
         } else {
-            throw new DataNotFoundException(String.format(categoryNotFound, id));
+            throw new DataNotFoundException(String.format("Category with id %d not found", id));
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java]: Why did you remove property in file and introduce String in code?

Comment (svelaf71) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java]: Because I would have to create lots of properties with different text and I thought that it's easier to write it as a String. Do I have to change it?

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java (hunk 3 above)
Comment (svelaf71) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CategoryController.java]: No need :D Will be fixed

**Comment by svelaf71** [comment]
Other changes I'll write it my next commit

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java (hunk 5 above)
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/CategoryService.java]: When we will apply localization it will be hardcoded message. So please move to the property file
